### PR TITLE
rgw: add expiration checking to avoid deleting object index by mistake

### PR DESCRIPTION
**Pull request by dongbula**
http://tracker.ceph.com/issues/21587
Signed-off-by: lvshuhua <lvshuhua@cmss.chinamobile.com>

### CODE REVIEW

**Comment by joscollin** [comment]
Jenkins retest this please.

@dongbula Fix the spelling `Fiexs:`

**Comment by dongbula** [comment]
@joscollin have repushed, thanks

File: src/rgw/rgw_rados.cc [hunk 1]
```diff
@@ -12773,6 +12773,7 @@ int RGWRados::cls_bucket_list(RGWBucketInfo& bucket_info, int shard_id, rgw_obj_
 
   map<string, bufferlist> updates;
   uint32_t count = 0;
+  utime_t now = ceph_clock_now();
```
Comment (mattbenjamin) [inline src/rgw/rgw_rados.cc]: is there a way we can avoid casting to utime_t?

Comment (dongbula) [reply src/rgw/rgw_rados.cc]: well,i will update other expiration prs recently,thx

Comment (dongbula) [reply src/rgw/rgw_rados.cc]: @mattbenjamin  changed it

**Comment by dongbula** [comment]
@joscollin @mattbenjamin help review when you are convenient，thanks

**Comment by dongbula** [comment]
@dang please help to review this pr, thx

**Comment by dongbula** [comment]
can we continue this pr? @mattbenjamin

**Comment by mattbenjamin** [comment]
@dongbula sure, thanks for updating, will reread

Matt

**Comment by dongbula** [comment]
@dang this error occurs when several rgws write data to a new bucket in multisite environment. Take a zonegroup contained two zones for example, upload data to a new bucket in master zone by several rgws, the first time syncing of bucket shard(slave zone) will trigger a list object op in master zone. The list op will call check_disk_state(RGWRados::cls_bucket_list) and may detect some errors for data delayed write back to disk. Then, cls_bucket_list will run some rm ops in cls_rgw_suggest_changes to clean these "abnormal" indexes. 
I have retested the current master version, this bug still exists. If you need more detail, please add more comment here.

**Comment by dongbula** [comment]
@dang Could you help to review this pr again, thanks

**Comment by joscollin** [comment]
jenkins retest this please

**Pull request closed without merging**